### ISSUE TITLE
When Tweakwise hits the feed URL a redirect might take place

### ISSUE THREAD

**Issue by dverkade** [Tweakwise/Magento2TweakwiseExport#42]
Issue Brief
--
When Tweakwise hits the feed URL a redirect might take place, causing invalid feeds to be retrieved. 

Environment
--
- [ ] PHP Version: 8.1
- [ ] Magento Version: 2.4.6-p4
- [ ] Magento Enterprise: No
- [ ] Tweakwise Export Version:
- [ ] Magento Deploy Mode: production

Steps to reproduce
--
1. Have a storeview on example.com and a separate backoffice URL, like backoffice.example.com
2. example.com runs on webserver 1 and backoffice.example.com runs on webserver 2
3. Crons only run on webserver 2 / the backoffice server. 
4. Feeds gets generated on webserver 2.

Actual result
--
1. When Tweakwise hits the feed URL it will redirect / fetch the feed from the store domain URL. I.e. https://backoffice.examaple.com/tweakwise/feed/export/key/1234 will get the feed from https://www.example.com/tweakwise/feed/export/key/4576 which might get this from another webserver which does not have (the latest ) feed. 

Expected result
--
List the expected results as a bullet list of expectations
- [ ] Do not redirect to store URL and fetch from the admin / backoffice domain directly.

**Comment by ahuininga-orisha** [comment]
@dverkade nvm, misunderstood the issue. I'll look into this and see if we can find an solution for this

**Comment by ahuininga-orisha** [comment]
@dverkade When you login to the admin is the admin url backoffice.example.com or do you have an custom path behind that? backoffice.exmple.com/admin_path

**Comment by dverkade** [comment]
@ah-net, yes it has a custom path for security purposes as the default path can easily be guessed

**Comment by ahuininga-orisha** [comment]
@dverkade And what happens if you go to backoffice.example.com without the admin path? Because for me it redirects to an store. So that means we don't do the redirect, magento does it, because backoffice.example.com is not an valid store/url
I'm not sure if this is something we can/should fix in that case.

**Comment by dverkade** [comment]
@ah-net, yes it's a valid store as backoffice.example.com is set on the default scope and the individual stores have other base URL's set in the store configuration.

**Comment by ahuininga-orisha** [comment]
@dverkade That explains it. Thx. I will see if i can reproduce that.

**Comment by ahuininga-orisha** [comment]
Thanks for the report. After investigation, the redirect is expected Magento behaviour — when the incoming request host doesn't match the store's configured base URL, Magento redirects to the correct domain.

The root cause here is that the Tweakwise backoffice feed URL is configured to point to `backoffice.example.com` instead of the store's frontend domain. The fix is to update the feed URL in the Tweakwise backoffice to `www.example.com/tweakwise/feed/export/...`.

For the feed to be accessible on that domain, either:
- Use shared storage (NFS, S3, etc.) so the file generated on the backoffice server is served by the frontend server, or
- Add an Nginx proxy rule on `www.example.com` that forwards `/tweakwise/feed/` requests internally to the backoffice server.

We'll close this as a configuration issue rather than a bug in the module. Feel free to reopen if there's a scenario we've missed.
